alf/algorithms/rlpd_algorithm.py

```diff
@@ -0,0 +1,213 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""RLPD Algorithm."""
+
+import torch
+from typing import Callable, Optional, Union
+
+import alf
+from alf.algorithms.config import TrainerConfig
+from alf.algorithms.sac_algorithm import SacAlgorithm, SacCriticState
+from alf.algorithms.sac_algorithm import ActionType, SacCriticInfo, SacInfo
+from alf.networks import ActorDistributionNetwork, CriticNetwork
+from alf.tensor_specs import TensorSpec, BoundedTensorSpec
+from alf.utils import common
+from alf.utils.schedulers import Scheduler
+
+
+@alf.configurable
+class RlpdAlgorithm(SacAlgorithm):
+    r"""RLPD algorithm, described in:
+
+    ::
+
+        Ball et al "Efficient Online Reinforcement Learning with Offline Data", arXiv:2302.02948
+
+    Currently, only continuous action spaces are supported.
+
+    In addition to the entropy regularization described in the SAC paper, we
+    also support KL-Divergence regularization if a prior actor is provided.
+    In this case, the training objective is:
+        :math:`E_\pi(\sum_t \gamma^t(r_t - \alpha D_{\rm KL}(\pi(\cdot)|s_t)||\pi^0(\cdot)|s_t)))`
+    where :math:`pi^0` is the prior actor.
+    """
+
+    def __init__(self,
+                 observation_spec,
+                 action_spec: BoundedTensorSpec,
+                 reward_spec=TensorSpec(()),
+                 actor_network_cls=ActorDistributionNetwork,
+                 critic_network_cls=CriticNetwork,
+                 repr_alg_ctor: Optional[Callable] = None,
+                 reward_weights=None,
+                 train_eps_greedy=1.0,
+                 epsilon_greedy=None,
+                 use_entropy_reward=True,
+                 normalize_entropy_reward=False,
+                 calculate_priority=False,
+                 num_critic_replicas=2,
+                 num_critic_targets=2,
+                 env=None,
+                 config: TrainerConfig = None,
+                 critic_loss_ctor=None,
+                 target_entropy=None,
+                 prior_actor_ctor=None,
+                 target_kld_per_dim=3.,
+                 initial_log_alpha=0.0,
+                 max_log_alpha=None,
+                 target_update_tau: Union[float, Scheduler] = 0.05,
+                 target_update_period: Union[int, Scheduler] = 1,
+                 parameter_reset_period: Union[int, Scheduler] = -1,
+                 dqda_clipping=None,
+                 actor_optimizer=None,
+                 critic_optimizer=None,
+                 alpha_optimizer=None,
+                 checkpoint=None,
+                 debug_summaries=False,
+                 name="RlpdAlgorithm"):
+        # **kwargs):
+        """
+        Refer to SacAlgorithm for more details for kwargs
+
+        Args:
+            name (str): The name of this algorithm.
+            num_critic_targets (int): Number of sampled subset of target critics
```

Comment:
``num_critic_targets``: maybe can change its name to reflect the ``sampled`` aspect? Currently, it is very close to the meaning of ``num_critic_replicas``

Reply:
Should ensure that ``num_critic_targets <= num_critic_replicas``.

Reply:
Updated.